dhall/src/Dhall/Parser/Token.hs

```diff
@@ -525,10 +528,13 @@ decOctet =
       where
         predicate c = '\x30' <= c && c <= '\x35'
 
-regName :: Parser Text
-regName = star (satisfy predicate <|> pctEncoded)
+domain :: Parser Text
+domain = star (domainLabel <> ".") <> domainLabel <> option "."
```

Comment:
I think this might fix the parsing failures you are running into:

```suggestion
domain = domainLabel <> star ("." <> domainLabel ) <> option "."
```

The issue is that the parser does not backtrack by default, so the `star` will fail to parse the last domain label and then not backtrack one iteration